code/modules/mob/living/simple_animal/hostile/necro.dm

```diff
@@ -96,11 +130,319 @@
 		ticker.mode.update_necro_icons_added(Controller)
 		ticker.mode.update_all_necro_icons()
 		ticker.mode.risen.Add(Controller)
-	name += " ([rand(1,1000)])"
 
-/mob/living/simple_animal/hostile/necro/copy/ListTargets()
-	. = ..()
-	return . - creator
+	if(name == initial(name) && !unique_name)
+		name += " ([rand(1,1000)])"
+
+/mob/living/simple_animal/hostile/necro/zombie/Life()
+	..()
+	/*TODO
+	First, check if the zombie can potentially evolve
+	Have the zombie move to a corpse and start chewing at it
+	If neither of these things are applicable, break some lights to set the mood
+	Otherwise, start wandering and bust down some doors to find more food
+	*/
+	if(!isUnconscious())
+		if(stance == HOSTILE_STANCE_IDLE && (!client)) //Not doing anything at the time
+			var/list/can_see = view(src, vision_range)
+			if(!busy)
+				if(can_evolve)//Can we evolve, and have we fed
+					busy = EVOLVING
+					check_evolve()
+					busy = 0
+				if((health < maxHealth) || (maxHealth < health_cap) && !busy)
+					var/mob/living/carbon/human/C = find_food(can_see)//Is there something to eat in range?
+					if(C) //If so, chow down
+						Goto(C, speed)
+						busy = MOVING_TO_TARGET
+						give_up(C) //If we're not there in 10 seconds, give up
+						if(C.Adjacent(src) && busy != EATING) //Once we've finally caught up
+							busy = EATING
+							eat(C)
+							C = null
+							walk(src, 0)
+				if(!busy && break_doors != CANT)//So we don't try to eat and open doors
+					var/obj/machinery/door/D = find_door(can_see)//Is there a door to open in range?
+					if(D)
+						if(D.Adjacent(src) && busy != OPENING_DOOR)
+							busy = OPENING_DOOR
+							force_door(D)
+							D = null
+							walk(src, 0)
+						else
+							Goto(D, speed)
+							busy = MOVING_TO_TARGET
+							give_up(D)
+		else
+			busy = 0
+			stop_automated_movement = 0
+	else
+		walk(src,0)
+
+/mob/living/simple_animal/hostile/necro/zombie/proc/find_food(var/list/can_see)
+	for(var/mob/living/carbon/human/C in can_see) //Because of how can_see lists things, it'll go in order of closest to furthest
+		if(C.isDead() && !busy && check_edibility(C))
+			return(C) //This would get the closest one
+
+/mob/living/simple_animal/hostile/necro/zombie/proc/find_door(var/list/can_see)
+	for(var/obj/machinery/door/D in can_see)
+		if(can_open_door(D) && !busy)
+			return(D)
+
+/mob/living/simple_animal/hostile/necro/zombie/proc/give_up(var/C)
+	spawn(100)
+		if(busy == MOVING_TO_TARGET)
+			if(C && !Adjacent(C))
+				C = null
```

Comment:
Previously it was entirely copypasted from giant spider code, which uses changing the attacking target to do things like cocoon objects in view, so the value of target was disused, so I opted to try and change it to clear out the variable being used in the proc beforehand.

You've got a point though, here it's doing nothing to set C = null, I'll rework it